extension/src/paymentHandler/klarna-make-payment.handler.js

```diff
@@ -114,25 +112,37 @@ function _createAdyenLineItemFromCustomLineItem(ctpLineItem, locales) {
   }
 }
 
-function _createShippingInfoAdyenLineItem(shippingInfo) {
+function _createShippingInfoAdyenLineItem(shippingInfo, locales) {
   return {
     id: `${shippingInfo.shippingMethodName}`,
     quantity: 1, // always one shipment item so far
-    description: _getShippingMethodDescription(shippingInfo),
+    description:
+      _getShippingMethodDescription(shippingInfo, locales) ||
+      KLARNA_DEFAULT_SHIPPING_METHOD_DESCRIPTION,
     amountIncludingTax: shippingInfo.price.centAmount,
     taxPercentage: shippingInfo.taxRate.amount * ADYEN_PERCENTAGE_MINOR_UNIT,
   }
 }
 
-function _getShippingMethodDescription(shippingInfo) {
-  const shippingMethod = shippingInfo.shippingMethod.obj
-  if (shippingMethod) return shippingMethod.description
+function _getShippingMethodDescription(shippingInfo, locales) {
+  const shippingMethod = shippingInfo.shippingMethod?.obj
+  if (shippingMethod) {
+    return _localizeOrFallback(
+      shippingMethod.localizedDescription,
+      locales,
+      shippingMethod.description
+    )
+  }
   return shippingInfo.shippingMethodName
 }
 
 function _localizeOrFallback(localizedString, locales, fallback) {
-  const locale = locales.find((l) => localizedString[l])
-  return locale ? localizedString[locale] : fallback
+  let result
```

Comment:
I guess the additional `result` variable is not needed, maybe it could be more readable without it.

Reply:
Do you mean to return immediately? I don't think it makes a difference here, so I'll just keep it as it is.